### PR TITLE
AzureKeyVault fetch mechanism is updated to chunks, refactoring (onetocny)

### PR DESCRIPTION
**Pull request by ozanovus**
**Task name**: AzureKeyVaultV1 and AzureKeyVaultV2

**Description**:  Paralellism is improved to prevent errors.

**Documentation changes required:** (Y/N) N

**Added unit tests:** (Y/N) Y

**Attached related issue:** (Y/N) N

**Checklist**:
- [X] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

**Comment by ozanovus** [comment]
@onetocny  FYI.

File: Tasks/AzureKeyVaultV2/package.json [hunk 1]
```diff
@@ -6,10 +6,10 @@
     "@types/node": "^10.17.0",
     "@types/q": "^1.5.0",
     "@types/uuid": "^8.3.0",
-    "azure-pipelines-task-lib": "2.8.0",
-    "azure-pipelines-tasks-azure-arm-rest-v2": "1.0.4",
-    "moment": "^2.29.4",
-    "vso-node-api": "6.0.1-preview"
+    "azure-devops-node-api": "11.2.0",
+    "azure-pipelines-task-lib": "^4.0.1-preview",
+    "azure-pipelines-tasks-azure-arm-rest-v2": "file:../../common-npm-packages/azure-arm-rest-v2/_build/azure-pipelines-tasks-azure-arm-rest-v2-3.215.0.tgz",
```
Comment (onetocny) [inline Tasks/AzureKeyVaultV2/package.json:11]: Same as in V1 - we need to wait for #17517 completion.

File: Tasks/AzureKeyVaultV1/operations/KeyVault.ts [hunk 2]
```diff
@@ -90,10 +90,13 @@ export class KeyVault {
         // Key vault task explicitly handles multi line masking - hence setting SYSTEM_UNSAFEALLOWMULTILINESECRET to true
         tl.setVariable("SYSTEM_UNSAFEALLOWMULTILINESECRET", "true");
 
-        if (downloadAllSecrets) {
-            return this.downloadAllSecrets(secretsToErrorsMap);
-        } else {
-            return this.downloadSelectedSecrets(this.taskParameters.secretsFilter, secretsToErrorsMap);
+        if (downloadAllSecrets)
+        {
```
Comment (ozanovus) [inline Tasks/AzureKeyVaultV1/operations/KeyVault.ts:94]: Could you please run format the document? (e.g. "alt+shift+f") 
Curly braces in TypeScript should be same level with "if" or similar conditions.

File: Tasks/AzureKeyVaultV1/operations/azure-arm-keyvault.ts [hunk 3]
```diff
@@ -28,49 +31,85 @@ export class KeyVaultClient extends azureServiceClient.ServiceClient {
...
+        {
+            return true;
         }
+
+        if (!error.message)
+        {
+            return false;
+        }
+
+        if (error.message.startsWith("Request timeout: "))
```
Comment (ozanovus) [inline Tasks/AzureKeyVaultV1/operations/azure-arm-keyvault.ts:102]: I think you can combine all true statements:

`(error.message?.startsWith("Request timeout: ") || error.message?.startsWith("getaddrinfo "))`

It shouldn't enter the condition if `error.message` is `null` or `undefined`

And you can remove `!error.message` condition in that case.

File: Tasks/AzureKeyVaultV1/operations/KeyVault.ts (hunk 2 above)
Comment (onetocny) [reply Tasks/AzureKeyVaultV1/operations/KeyVault.ts]: Done

File: Tasks/AzureKeyVaultV1/operations/azure-arm-keyvault.ts (hunk 3 above)
Comment (onetocny) [reply Tasks/AzureKeyVaultV1/operations/azure-arm-keyvault.ts]: Unfortunately `?.` operator is available from Node 14 (see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#browser_compatibility). However, I have introduced `retriableErrorMessages` array and check `error.message` against it. Thus the conditions here are simplified.

**Review by onetocny — APPROVED** [review]

**Pull request merged**